src/table/utils/handle-key-press.ts

```diff
@@ -76,26 +102,41 @@ export const arrowKeysNavigation = (evt, rowAndColumnCount, cellCoord, topAllowe
   return [nextRow, nextCol];
 };
 
-export const getRowAndColumnCount = (rootElement) => {
-  const rowCount = rootElement.getElementsByClassName('sn-table-row').length;
-  const columnCount = rootElement.getElementsByClassName('sn-table-head-cell').length;
-
-  return { rowCount, columnCount };
-};
-
-export const moveFocus = (evt, rootElement, cellCoord, setFocusedCellCoord, topAllowedRow) => {
+/**
+ * resets and adds new focus to a table cell based which key is pressed
+ */
+export const moveFocus = (
+  evt: React.KeyboardEvent,
+  rootElement: HTMLElement,
+  cellCoord: [number, number],
+  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>,
+  topAllowedRow?: number
+) => {
   preventDefaultBehavior(evt);
-  evt.target.setAttribute('tabIndex', '-1');
-  const rowAndColumnCount = getRowAndColumnCount(rootElement);
-  const nextCellCoord = arrowKeysNavigation(evt, rowAndColumnCount, cellCoord, topAllowedRow);
+  (evt.target as HTMLElement).setAttribute('tabIndex', '-1');
+  const nextCellCoord = getNextCellCoord(evt, rootElement, cellCoord, topAllowedRow);
   const nextCell = getCellElement(rootElement, nextCellCoord);
   updateFocus({ focusType: 'focus', cell: nextCell });
   setFocusedCellCoord(nextCellCoord);
 
   return nextCell;
 };
 
-export const headHandleKeyPress = ({
+interface handleHeadProps {
+  evt: React.KeyboardEvent;
+  rootElement: HTMLElement;
+  cellCoord: [number, number];
+  column: Column;
+  changeSortOrder: (layout: TableLayout, column: Column) => Promise<void>;
```

Comment:
changeSortOrder used in both handleHeadProps and RenderWithCarbonArguments. We can extract it to be ChangeSortOrder.

Reply:
done